### PR TITLE
Add management API endpoint for listing trades

### PR DESCRIPTION
**Pull request by ymasiuk**

### CODE REVIEW

File: app/api/v2/management/entities/trade.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# encoding: UTF-8
+# frozen_string_literal: true
+
+module API
+  module V2
+    module Management
+      module Entities
+        class Trade < Base
```
Comment (ysv) [inline app/api/v2/management/entities/trade.rb:8]: I suggest to return all Trade columns
And if you don't need something filter it in applogic side

**Review by ysv — CHANGES_REQUESTED** [review]

**Comment by ysv** [comment]
@ymasiuk squash commits with the next title
`Add management API endpoint for listing trades`

**Comment by ymasiuk** [comment]
@ysv updated

File: app/api/v2/management/trades.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+          member = Member.find_by!(uid: params[:uid]) if params[:uid].present?
+
+          Trade
+            .order(id: :desc)
+            .includes(:market)
+            .tap { |q| q.where!(market: market) if market }
+            .tap { |q| q.where!("ask_member_id = #{member.id} OR bid_member_id = #{member.id}") if member }
+            .page(params[:page])
+            .per(params[:limit])
+            .tap { |q| present q, with: API::V2::Management::Entities::Trade }
```
Comment (ysv) [inline app/api/v2/management/trades.rb:32]: add includes with members for avoiding n+1 and remove market because we don't need it

**Review by ysv — CHANGES_REQUESTED** [review]

File: app/api/v2/management/trades.rb (hunk 2 above)
Comment (ysv) [reply app/api/v2/management/trades.rb:32]:
```ruby
          Trade
            .order(id: :desc)
            .includes(:ask_member, :bid_member)
            .tap { |q| q.where!(market: market) if market }
            .tap { |q| q.where!("ask_member_id = #{member.id} OR bid_member_id = #{member.id}") if member }
            .page(params[:page])
            .per(params[:limit])
            .tap { |q| present q, with: API::V2::Management::Entities::Trade }
```

**Review by ysv — APPROVED** [review]

**Review by mod — APPROVED** [review]

**Pull request merged**